astropy/io/votable/tests/table_test.py

```diff
@@ -10,17 +10,47 @@
 import pytest
 
 from astropy.config import reload_config, set_temp_config
-from astropy.io.votable import conf, tree, validate
+from astropy.io.votable import conf, from_table, is_votable, tree, validate
 from astropy.io.votable.exceptions import E25, W39, VOWarning
 from astropy.io.votable.table import parse, writeto
 from astropy.table import Column, Table
 from astropy.table.table_helpers import simple_table
 from astropy.units import Unit
-from astropy.utils.data import get_pkg_data_filename, get_pkg_data_fileobj
+from astropy.utils.data import (get_pkg_data_filename, get_pkg_data_fileobj,
+        get_pkg_data_path)
 from astropy.utils.exceptions import AstropyDeprecationWarning
 from astropy.utils.misc import _NOT_OVERWRITING_MSG_MATCH
 
 
+@pytest.fixture
+def home_is_data(monkeypatch):
+    """
+    Pytest fixture to run a test case with tilde-prefixed paths.
+
+    In the tilde-path case, environment variables are temporarily
+    modified so that '~' resolves to the data directory.
+    """
+    path = get_pkg_data_path('data')
+    # For Unix
+    monkeypatch.setenv('HOME', path)
+    # For Windows
+    monkeypatch.setenv('USERPROFILE', path)
+
+
+@pytest.fixture
+def home_is_tmpdir(monkeypatch, tmpdir):
```

Comment:
`tmpdir` uses a third party library, whereas the `tmp_path` fixture uses Python standard library. Using `tmpdir` in new code is therefore discouraged. See also the note at https://docs.pytest.org/en/stable/how-to/tmp_path.html#the-tmpdir-and-tmpdir-factory-fixtures

Reply:
Ah, thanks for the pointer!